src/modules/Elsa.Workflows.Management/Filters/WorkflowInstanceFilter.cs

```diff
@@ -106,19 +111,47 @@ public IQueryable<WorkflowInstance> Apply(IQueryable<WorkflowInstance> query)
-        if (filter.ToCreatedAt is not null) query = query.Where(x => filter.ToCreatedAt >= x.CreatedAt);
-        if (filter.FromCreatedAt is not null) query = query.Where(x => filter.FromCreatedAt <= x.CreatedAt);
+        if (filter.HasIncidents != null) query = filter.HasIncidents == true ? query.Where(x => x.IncidentCount > 0) : query.Where(x => x.IncidentCount == 0);
+
+        if (TimestampFilters != null)
+        {
+            foreach (TimestampFilter timestampFilter in TimestampFilters)
+            {
+                var column = timestampFilter.Column;
+                var timestamp = timestampFilter.Timestamp;
+                var isZeroTime = timestamp.TimeOfDay == TimeSpan.Zero;
+                var startDay = new DateTimeOffset(timestamp.Date);
+                var endDay = startDay.AddDays(1);
+
+                query = timestampFilter.Operator switch
+                {
+                    TimestampFilterOperator.Is when isZeroTime => query.Where($"{column} >= @0 && {column} < @1", startDay, endDay),
+                    TimestampFilterOperator.Is => query.Where($"{column} == @0", timestamp),
+                    TimestampFilterOperator.IsNot when isZeroTime => query.Where($"{column} < @0 || {column} >= @1", startDay, endDay),
+                    TimestampFilterOperator.IsNot => query.Where($"{column} != @0", timestamp),
+                    TimestampFilterOperator.GreaterThan when isZeroTime => query.Where($"{column} > @0", endDay),
+                    TimestampFilterOperator.GreaterThan => query.Where($"{column} > @0", timestamp),
+                    TimestampFilterOperator.GreaterThanOrEqual when isZeroTime => query.Where($"{column} >= @0", startDay),
+                    TimestampFilterOperator.GreaterThanOrEqual => query.Where($"{column} >= @0", timestamp),
+                    TimestampFilterOperator.LessThan when isZeroTime => query.Where($"{column} < @0", startDay),
+                    TimestampFilterOperator.LessThan => query.Where($"{column} < @0", timestamp),
+                    TimestampFilterOperator.LessThanOrEqual when isZeroTime => query.Where($"{column} <= @0", endDay),
+                    TimestampFilterOperator.LessThanOrEqual => query.Where($"{column} <= @0", timestamp),
+                    _ => query
+                };
+            }
+        }
 
         var searchTerm = filter.SearchTerm;
         if (!string.IsNullOrWhiteSpace(searchTerm))
         {
             query =
                 from instance in query
                 where instance.Name!.Contains(searchTerm)
-                      || instance.DefinitionVersionId.Equals(searchTerm)
-                      || instance.Id.Equals(searchTerm)
-                      || instance.DefinitionId.Equals(searchTerm)
-                      || instance.CorrelationId!.Equals(searchTerm)
+                      || instance.DefinitionVersionId.Contains(searchTerm)
```

Comment:
Should we only be able to search on an (part of) id or do we also want to be able to search on the name of the workflow definition?

Reply:
It would be nice to be able to also search on a part of the workflow definition name, but I wasn't sure if it is worth the added complexity to achieve that. I think that it would require to do a preliminary search on the workflow definitions, then selecting the matching definition IDs, and adding those to the workflow instance filter. If you think it is worthwhile, we can do it. It would certainly offer a very nice experience to the user.

Reply:
A version where we can reduce the complexity is where we can use the search in the workflow definition selection box and use that to select the definitions we are interested in.

Reply:
Aha - so basically grab a set of matching workflow definition IDs when the user enters a search term in the text field, and including that as part of the request. That's pretty smart! 😎 I will send another PR with that improvement 👍🏻